doc/compiler/memory/MemoryManager.md

```diff
@@ -0,0 +1,214 @@
+# OMR Compiler Memory Manager
+
+The OMR Compiler doesn't necessarily operate in a vacuum, but rather 
+as one component of a Managed runtime. Additionally,  communications 
+between components could be extern "C". Therefore, one can't simply 
+use something like `::operator new` since it can throw. Additionally, 
+mutli-threaded performance of malloc historically has been hit or miss, 
+and any generic common allocator would have poor interaction between 
+components, with little to no insight into usage.
+
+Using a domain-specialized common allocator, such as `omrmem_allocate_memory` 
+in the Port Library, is expensive (both in terms of memory footprint 
+and execution time) for many of the types of allocations performed 
+by the Compiler; there are too many participants for informal trust, 
+book-keeping is expensive, and optimizing for divergent component 
+use-cases is difficult to impossible.
+
+Therefore, the compiler manages memory its own memory.
+
+
+## Compiler Memory Manager Hierarchy
```

Comment:
IMO, leading this description with an [ASCII-art diagram](http://asciiflow.com) of the hierarchies would be a great choice.